experiment/compatibility-versions/e2e-k8s-compatibility-versions.sh

```diff
@@ -327,10 +328,16 @@ main() {
   res=0
   create_cluster || res=$?
 
-  # Clone the specific Kubernetes release branch
-  clone_kubernetes_release
+  # Clone the previous versions Kubernetes release branch
+  # TODO(aaron-prindle) extend the branches to test from n-1 -> n-1..3 as more k8s releases are done that support compatibility versions
+  export PREV_RELEASE_BRANCH="release-${EMULATED_VERSION}"
+  git clone --single-branch --branch "${PREV_RELEASE_BRANCH}" https://github.com/kubernetes/kubernetes.git "${PREV_RELEASE_BRANCH}"
```

Comment:
Do we really need another full clone? (also if we're going to clone, we should add some performance related flags, cloning kubernetes in full is really expensive with the 10 years of history)

If we need to use git, let's use a worktree. But I think we can fetch binaries for the old release and for the new release we should already have the checkout in order, presumably.

Reply:
+1, I'm familiar with how to pull branch with minimal history (`--depth=1`) which I've changed this to use now.  Can you explain how a worktree would help in this context or give an example for what you are suggesting with `If we need to use git, let's use a worktree.`?  Thanks

Reply:
> +1, I'm familiar with some of the flags to pull branch with minimal history (--depth=1, --single-branch) which I've changed this to use now. Did you have any other suggestions for flags there?

`--depth=1` will break `git describe`, in prow we use `--filter=blob:none` instead

Reply:
Ah got it, I've updated this to use `--filter=blob:none` now:
```
  git clone --filter=blob:none --single-branch --branch "${PREV_RELEASE_BRANCH}" https://github.com/kubernetes/kubernetes.git "${PREV_RELEASE_BRANCH}"

```